docs/sphinx/source/authenticating-to-vespa-cloud.ipynb

```diff
@@ -0,0 +1,984 @@
+      "Setting target cloud...\n",
+      "Running: vespa config set target cloud\n",
+      "\n",
+      "Api-key found for control plane access. Using api-key.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from vespa.deployment import VespaCloud\n",
+    "from vespa.application import Vespa\n",
+    "import os\n",
+    "\n",
+    "# Key is only used for CI/CD. Can be removed if logging in interactively\n",
+    "key = os.getenv(\"VESPA_TEAM_API_KEY\", None)\n",
+    "if key is not None:\n",
+    "    key = key.replace(r\"\\n\", \"\\n\")  # To parse key correctly\n",
+    "\n",
+    "\n",
+    "vespa_cloud = VespaCloud(\n",
+    "    tenant=tenant_name,  # Note that the name cannot contain the characters `-` or `_`.\n",
+    "    application=application,\n",
+    "    key_content=key,  # Prefer to use  key_location=\"<path-to-key-file.pem>\"\n",
+    "    application_package=package,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dec50045",
+   "metadata": {},
+   "source": [
+    "When you have authenticated to the control-plane of Vespa Cloud, key/cert for data-plane authentication will be generated automatically for you, if none exists.\n",
+    "\n",
+    "The `data-plane-public-cert.pem` will be added to the application package (in `/security/clients.pem` directory) that will be deployed. You should keep them safe, as any app or users that need data-plane access to your Vespa application will need them.\n",
+    "\n",
+    "For `dev`-deployments, we allow redeploying an application with a different key/cert than the previous deployment. For `prod`-deployments however, this is not allowed, and will require a `validation-overrides`-specification in the application package.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "careful-savage",
+   "metadata": {},
+   "source": [
+    "## Deploy to Vespa Cloud\n",
+    "\n",
+    "The app is now defined and ready to deploy to Vespa Cloud.\n",
+    "\n",
+    "Deploy `package` to Vespa Cloud, by creating an instance of\n",
+    "[VespaCloud](https://pyvespa.readthedocs.io/en/latest/reference-api.html#vespa.deployment.VespaCloud):\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "197c0a27",
+   "metadata": {},
+   "source": [
+    "The following will upload the application package to Vespa Cloud Dev Zone (`aws-us-east-1c`), read more about [Vespa Zones](https://cloud.vespa.ai/en/reference/zones.html).\n",
+    "The Vespa Cloud Dev Zone is considered as a sandbox environment where resources are down-scaled and idle deployments are expired automatically.\n",
+    "For information about production deployments, see the following [docs](https://cloud.vespa.ai/en/reference/deployment).\n",
+    "\n",
+    "> Note: Deployments to dev and perf expire after 7 days of inactivity, i.e., 7 days after running deploy. This applies to all plans, not only the Free Trial. Use the Vespa Console to extend the expiry period, or redeploy the application to add 7 more days.\n"
```

Comment:
I think this is 14 days now

Reply:
... until credits is launched ... :-)

Reply:
Changed it now. Will keep a note to self to change when credits is launched.